sakai-evaluation-tool/src/java/org/sakaiproject/evaluation/tool/reporting/XLSReportExporter.java

```diff
@@ -79,223 +95,627 @@ public void setMessageLocator(MessageLocator locator) {
-                    //skip instructor items that aren't for the current user
-                    continue;
-                }
-                
-                // 7) check answersmap for an answer, if there put in cell, if missing, insert blank
-                EvalAnswer answer = dti.getAnswer(responseId);
-                Cell responseCell = row.createCell(dtiCounter);
-                // In Eval, users can leave questions blank, in which case this will be null
-                if (answer != null) {
-                    setPlainStringCell(responseCell, responseAggregator.formatForSpreadSheet(answer.getTemplateItem(), answer));
+        // Convert the map structure of instructor responses into a List<List<String>>, representing rows of data
+        List<List<String>> instructorRelatedResponses = new ArrayList<List<String>>();
+        for( Long responseID : answerMap.keySet() )
+        {
+            // Loop through the instructors for the current response
+            for( User instructor : answerMap.get( responseID ).keySet() )
+            {
+                // Dump the data into a list of strings representing a spreadsheet row (so it can be sorted)
+                List<String> row = new ArrayList<String>();
+                row.add( SECTION_OR_SITE_COLUMN_NUM, "" );
+                row.add( RESPONSE_ID_COLUMN_NUM, responseID.toString() );
+                row.add( INSTRUCTOR_ID_COLUMN_NUM, "" );
+                row.add( INSTRUCTOR_FIRST_NAME_COLUMN_NUM, "" );
+                row.add( INSTRUCTOR_LAST_NAME_COLUMN_NUM, "" );
+                if( instructor != null )
+                {
+                    row.set( INSTRUCTOR_ID_COLUMN_NUM, instructor.getDisplayId() );
+                    row.set( INSTRUCTOR_FIRST_NAME_COLUMN_NUM, instructor.getFirstName() );
+                    row.set( INSTRUCTOR_LAST_NAME_COLUMN_NUM, instructor.getLastName() );
                 }
-                if (dti.usesComments()) {
-                    // put comment in the extra column
-                    dtiCounter++;
-                    setPlainStringCell(row.createCell(dtiCounter), 
-                            (answer == null || EvalUtils.isBlank(answer.getComment())) ? "" : answer.getComment());
+
+                int questionCounter = 0;
+                for( EvalAnswer answer : answerMap.get( responseID ).get( instructor ) )
+                {
+                    row.set( SECTION_OR_SITE_COLUMN_NUM, responseAggregator.getCommaSeparatedGroupNames( new String[] { answer.getResponse().getEvalGroupId() } ) );
+                    row.add( QUESTION_COMMENTS_COLUMN_START_INDEX_INSTRUCTOR_SHEET + questionCounter, "" );
+                    if( answer != null )
+                    {
+                        row.set( QUESTION_COMMENTS_COLUMN_START_INDEX_INSTRUCTOR_SHEET + questionCounter, responseAggregator.formatForSpreadSheet( answer.getTemplateItem(), answer ) );
+                    }
```

Comment:
move next if statement into this block
String comment = StringUtils.trimToEmpty(answer.getComment());
if (!comment.isEmpty()) row.add(......, comment)

also the inline prefix increment on questionCounter is hard to read in this context